src/Core/Checkout/Customer/SalesChannel/RegisterConfirmRoute.php

```diff
@@ -101,11 +102,14 @@ public function confirm(RequestDataBag $dataBag, SalesChannelContext $context):
         }
 
         $this->validator->validate(
-            ['em' => $dataBag->get('em')],
+            [
+                'em' => $dataBag->get('em'),
+                'doubleOptInRegistration' => $customer->getDoubleOptInRegistration(),
+            ],
             $this->getBeforeConfirmValidation(hash('sha1', $customer->getEmail()))
         );
 
-        if ($customer->getActive()) {
+        if ($customer->getActive() || $customer->getDoubleOptInConfirmDate() !== null) {
```

Comment:
Please add `tests/unit/php/Core/Checkout/Customer/SalesChannel/RegisterConfirmRouteTest.php` with unit tests. Assert that the exception is thrown and the validation definition includes `doubleOptInRegistration`.

Reply:
added...